### PR TITLE
add retry to pool query

### PR DESCRIPTION
**Pull request by piepiper**
add retry to pool query in case of mysql server keepalive connection timeout.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -161,6 +161,9 @@ By default it returns query result in dictionary form.
     if specified and is False, `pool.query()` return query result in list
     form.
 
+-   `retry`:
+    try to request db for another N times after OperationalError. default: N = 3
+
```
Comment (drmingdrmer) [inline README.md]: 上面的syntax部门没改啊...

Comment (drmingdrmer) [inline README.md]: default 设置成0, 不retry.

Comment (drmingdrmer) [inline README.md]: 写操作重发会有问题.

File: mysqlconnpool.py [hunk 2]
```diff
@@ -110,10 +110,19 @@ def conn_query(conn, sql, use_dict=True):
     return rst
 
 
-def query(pool, sql, use_dict=True):
+def query(pool, sql, use_dict=True, retry=3):
 
-    with pool() as conn:
-        return conn_query(conn, sql, use_dict=use_dict)
+    for i in xrange(0, retry):
```
Comment (drmingdrmer) [inline mysqlconnpool.py]: 注意retry=0的行为.

File: mysqlconnpool.py [hunk 3]
```diff
@@ -110,10 +110,19 @@ def conn_query(conn, sql, use_dict=True):
...
+        try:
+            with pool() as conn:
+                return conn_query(conn, sql, use_dict=use_dict)
+        except MySQLdb.OperationalError as e:
+            logger.warn(repr(e) + " conn_query error {sql}".format(sql=sql))
+            continue
+        else:
+            raise
+    else:
+        raise
```
Comment (drmingdrmer) [inline mysqlconnpool.py]: 加测试..

**Comment by piepiper** [comment]
Function had been added by xp. PR closed.

**Pull request closed without merging**